tasks.py

```diff
@@ -9,56 +9,66 @@
 from tempfile import NamedTemporaryFile
 import csv
 import os
+import sys
 
 
 DONE = " ✅"
 
 
-def list():
+def list(stdout=sys.stdout, filename="tasks.csv"):
     """
     List the current known tasks.
     """
 
-    with open("tasks.csv") as tasks_file:
-        reader = csv.reader(tasks_file)
-        for name, completed in reader:
-            print(f"{name}{DONE if completed else ''}")
+    try:
+        with open(filename) as tasks_file:
+            reader = csv.reader(tasks_file)
+            for name, completed in reader:
+                stdout.write(f"{name}{DONE if completed == 'True' else ''}\n")
+    except FileNotFoundError:
+        stdout.write("There are no tasks.\n")
 
 
-def create(name):
+def create(name, status=False, filename="tasks.csv"):
     """
     Create a new task.
     """
 
-    with open("tasks.csv", "a") as tasks_file:
+    with open(filename, "a") as tasks_file:
         writer = csv.writer(tasks_file)
-        writer.writerow([name, False])
+        writer.writerow([name, status])
 
 
-def complete():
+def complete(completing=True, stdout=sys.stdout, filename="tasks.csv"):
     """
     Mark an existing task as completed.
     """
 
-    with (
-        open("tasks.csv") as tasks_file,
-        NamedTemporaryFile("w", delete=False) as new,
-    ):
-        reader = csv.reader(tasks_file)
-        print("Current tasks:")
-        for id, (name, completed) in enumerate(reader):
-            print(id, name, completed)
-
-        to_complete = int(input("task ID?> "))
-        writer = csv.writer(new)
-        tasks_file.seek(0)
-        for id, (name, completed) in enumerate(reader):
-            if id == to_complete:
-                writer.writerow([name, True])
-            else:
-                writer.writerow([name, completed])
-
-    os.rename(new.name, "tasks.csv")
+    try:
+        with (
+            open(filename) as tasks_file,
+            NamedTemporaryFile("w", delete=False) as new,
+        ):
+            reader = csv.reader(tasks_file)
+            stdout.write("Current tasks:\n")
+            for id, (name, completed) in enumerate(reader):
+                if completed != 'True':
+                    stdout.write(f"{id + 1} {name} {completed}\n")
```

Comment:
Nice -- enumerate also takes a second argument, the start number, so you can also have fixed this via enumerate(reader, 1) rather than i + 1.